### PR TITLE
Added DRONE_RUNNER_VOLUMES argument

### PR DESCRIPTION
**Pull request by devkimittal**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -10,9 +10,10 @@ import (
 
 // Config provides the system configuration.
 type Config struct {
-	Debug      bool   `envconfig:"DRONE_DEBUG"`
-	Trace      bool   `envconfig:"DRONE_TRACE"`
-	ServerName string `envconfig:"SERVER_NAME" default:"drone"`
+	Debug                bool     `envconfig:"DRONE_DEBUG"`
+	Trace                bool     `envconfig:"DRONE_TRACE"`
+	ServerName           string   `envconfig:"SERVER_NAME" default:"drone"`
+	DRONE_RUNNER_VOLUMES []string `envconfig:"DRONE_RUNNER_VOLUMES"`
```
Comment (vistaarjuneja) [inline config/config.go]: Name the variable `Volumes` and change the env variable to `CI_MOUNT_VOLUMES` to make it consistent with the other runners. Also, create another section for `Runner` and move this there.

File: handler/setup.go [hunk 2]
```diff
@@ -95,6 +96,7 @@ func HandleSetup() http.HandlerFunc {
 
 		// fmt.Printf("setup request config: %+v\n", s.SetupRequestConfig)
 		s.Volumes = append(s.Volumes, getSharedVolume())
+		s.Volumes = appendConfigVolume(s.Volumes)
```
Comment (vistaarjuneja) [inline handler/setup.go]: let the function return the volumes from the config, and do the appending here. Appending a list in different functions makes it a bit hard to follow

File: handler/setup.go [hunk 3]
```diff
@@ -200,6 +202,22 @@ func getDockerSockVolume() *spec.Volume {
 	}
 }
 
+func appendConfigVolume(volumes []*spec.Volume) []*spec.Volume {
+	loadedConfig, _ := config.Load()
```
Comment (vistaarjuneja) [inline handler/setup.go]: Pass in this config to the HandleSetup / HandleStep handlers when they are loaded, so that we don't have to load from env variables for each call

File: handler/step.go [hunk 4]
```diff
@@ -110,6 +112,19 @@ func getSharedVolumeMount() *spec.VolumeMount {
 	}
 }
 
+func appendConfigVolumeMount(volumes []*spec.VolumeMount) []*spec.VolumeMount {
+	loadedConfig, _ := config.Load()
```
Comment (vistaarjuneja) [inline handler/step.go]: Ditto for the config load

File: handler/setup.go [hunk 5]
```diff
@@ -200,6 +202,22 @@ func getDockerSockVolume() *spec.Volume {
...
+	droneRunnerVolumes := loadedConfig.DRONE_RUNNER_VOLUMES
+	for _, droneRunnerVolume := range droneRunnerVolumes {
+		volume := &spec.Volume{
+			HostPath: &spec.VolumeHostPath{
+				Name: strings.Split(droneRunnerVolume, ":")[0],
```
Comment (vistaarjuneja) [inline handler/setup.go]: Create another function to parse the volume so the logic is easy to see - similar to https://github.com/drone-runners/drone-runner-aws/pull/88/files#diff-3c718f8d6204bf9137ea1ed7872d05de927fb5ef0d8d082f4fe5df62cd0c3600R11

File: handler/setup.go [hunk 6]
```diff
@@ -200,6 +202,22 @@ func getDockerSockVolume() *spec.Volume {
...
+				Path: strings.Split(droneRunnerVolume, ":")[0],
+				ID:   random(),
```
Comment (vistaarjuneja) [inline handler/setup.go]: Make sure this is able to handle the case where we want the same local file mounted on different paths in the containers - eg:
`/path/to/a.txt:/path/to/b.txt,/path/to/a.txt:/container/path/to/a.txt,...`
You can have a look as to how it's done here if the above case is not working -  https://github.com/drone-runners/drone-runner-aws/pull/88/files - this should handle all the cases

Comment (devkimittal) [reply handler/setup.go]: yeah made sure of it

File: handler/setup.go [hunk 7]
```diff
@@ -200,6 +202,34 @@ func getDockerSockVolume() *spec.Volume {
 	}
 }
 
+func appendConfigVolume(config *config.Config) []*spec.Volume {
```
Comment (vistaarjuneja) [inline handler/setup.go]: Rename the function to `getGlobalVolumes`

File: handler/step.go [hunk 8]
```diff
@@ -110,6 +113,31 @@ func getSharedVolumeMount() *spec.VolumeMount {
 	}
 }
 
+func appendConfigVolumeMount(config *config.Config) []*spec.VolumeMount {
```
Comment (vistaarjuneja) [inline handler/step.go]: Ditto

File: handler/setup.go (hunk 7 above)
Comment (devkimittal) [reply handler/setup.go]: oh yes, pushed it

**Review by vistaarjuneja — APPROVED** [review]

**Comment by vistaarjuneja** [comment]
LGTM, thanks for addressing the comments.
